### PR TITLE
Dgj 488 v406 merge latest

### PR DESCRIPTION
**Pull request by david-sabadash**
## Summary

Merges in changes from the forms-flow-ai v4.0.6 branch, ensures our custom implementations are applied overtop

## Changes

new files and updates from forms-flow-ai 4.0.6

### CODE REVIEW

File: forms-flow-api/src/formsflow_api/resources/application.py [hunk 1]
```diff
@@ -119,17 +124,32 @@ def get():
...
     @profiletime
-    def get(application_id):
-        """Get application by id."""
+    def get(application_id: int):
+        """Get application by id.
+
+        : application_id:- List the application for particular application_id
+        """
         try:
             if auth.has_role([REVIEWER_GROUP]):
```
Comment (iman-jamali-fw) [inline forms-flow-api/src/formsflow_api/resources/application.py:154]: We added a role, "manager", for the supervisors. May need to update this if planned to return applications for that role in the future. Don't think have an issue at the moment.

File: forms-flow-api/src/formsflow_api/resources/user.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            url_path = f"users?username={user_name}"
+        response = self.client.get_request(url_path=url_path)
+        response = response[0]
+        if response is None:
+            return {"message": "User not found"}, HTTPStatus.NOT_FOUND
+        return response
+
+    @auth.require
+    @profiletime
+    def put(self) -> dict:
```
Comment (iman-jamali-fw) [inline forms-flow-api/src/formsflow_api/resources/user.py:43]: We may want to review this endpoint to make sure it does not have security implications. It'll be interesting to see what fields a user can update about his/her profile. Whether email/IDIR or user group/role are one of them?

We can also disable it until there is a need for it.

File: forms-flow-forms/package.json [hunk 3]
```diff
@@ -17,63 +17,63 @@
...
-    "mocha": "^9.1.3",
+    "mocha": "^9.2.2",
     "nodemon": "^2.0.14",
     "pre-commit": "^1.2.2",
     "superagent-use": "^0.1.0",
-    "supertest": "^6.1.6",
+    "supertest": "^6.2.2",
     "supertest-capture-error": "^1.0.0"
   },
   "directories": {
```
Comment (iman-jamali-fw) [inline forms-flow-forms/package.json:79]: This should help get rid of the security vulnerability warnings we get on the repo :)

File: forms-flow-web/src/components/Form/Item/Edit.js [hunk 4]
```diff
@@ -223,6 +223,12 @@ const Edit = React.memo((props) => {
     
   };
 
+// setting the main option details to the formdata
+  const handleChange = (path, event) => {
+    const {target} = event;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    dispatchFormAction({type: path, value});
+  
   // setting the main option details to the formdata
```
Comment (iman-jamali-fw) [inline forms-flow-web/src/components/Form/Item/Edit.js:232]: Seems like `handleChange` is duplicated.

**Review by iman-jamali-fw — COMMENTED** [review]
Finished the review. Huge work, great job @david-sabadash! 

I had a few comments, if you could please take a look.

If you could, please write up the upgrade process, much value in that.

Thanks greatly

File: forms-flow-web/src/components/Form/Item/Edit.js (hunk 4 above)
Comment (david-sabadash) [reply forms-flow-web/src/components/Form/Item/Edit.js:232]: good catch, change is added to my current commit for fixing build issues

File: forms-flow-api/src/formsflow_api/resources/application.py (hunk 1 above)
Comment (david-sabadash) [reply forms-flow-api/src/formsflow_api/resources/application.py:154]: I'm not seeing any mention of the 'Manager' role in this file on other branches. I'll look for where to re-add it though.

Comment (simensma-fresh) [reply forms-flow-api/src/formsflow_api/resources/application.py:154]: @david-sabadash @iman-jamali-fw We should keep this as the `REVIEWER` group. The manager role does not have any additional permissions

**Review by iman-jamali-fw — APPROVED** [review]

**Pull request merged**